### PR TITLE
Config: Add video transcoding resolution option

### PR DESCRIPTION
**Pull request by lukasdotcom**
This is a starting point for work on #3466. This update will add a configuration variable of PHOTOPRISM_FFMPEG_RESOLUTION that will let you set a maximum resolution for videos. This allows you to save disk space and more importantly bandwidth when watching videos in the browser.

Acceptance Criteria:

- [ ] Features and enhancements must be fully implemented so that they can be released at any time without additional work
- [ ] Automated unit and/or acceptance tests are mandatory to ensure the changes work as expected and to reduce repetitive manual work
- [ ] Frontend components must be responsive to work and look properly on phones, tablets, and desktop computers; you must have tested them on all major browsers and different devices
- [ ] Documentation and translation updates should be provided if needed
- [ ] In case you submit database-related changes, they must be tested and compatible with SQLite 3 and MariaDB 10.5.12+

### CODE REVIEW

**Comment by lastzero** [comment]
@Lukasdotcom Thank you very much for your contribution! This looks good to me at first sight. Of course, it should still be very well tested, as video transcoding is a core feature that users expect to work at all times. Let us know when you have questions or think it's ready to be merged so users can test it with our development preview build.

File: internal/ffmpeg/convert.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func AvcConvertCommand(fileName, avcName string, opt Options) (result *exec.Cmd,
 			"-i", fileName,
 			"-movflags", "faststart",
 			"-pix_fmt", "yuv420p",
+			// "-vf", "scale='-2:trunc("+opt.Resolution+"/2)*2'",
```
Comment (lukasdotcom) [inline internal/ffmpeg/convert.go:28]: Should I have the resolution limit also apply to animated images or only to videos?

Comment (lastzero) [reply internal/ffmpeg/convert.go:28]: That's a great question we should actually ask our users...

**Comment by lukasdotcom** [comment]
This should now work well enough for testing.
I just finished testing this with software encoding and apple's hardware encoding.

**Comment by lastzero** [comment]
If you give me a thumbs up as confirmation, I'll be happy to merge your changes for further testing!

**Comment by lukasdotcom** [comment]
👍

**Comment by lastzero** [comment]
@Lukasdotcom Would it make sense to specify the resolution as the longest side, depending on the aspect ration, rather than the height only? For images, it's typically done that way, but for videos, I'm not 100% sure. Phones for example often show videos in portrait mode.

Feel free to make suggestions on this or provide a follow-up PR. We could also rename RESOLUTION to HEIGHT, for example.

**Comment by lastzero** [comment]
With FFmpeg it seems to be possible to keep the aspect ratio and specify only the longest side:
```
ffmpeg -i input.jpg -vf scale=320:-1 output_320.png
```

Docs: http://trac.ffmpeg.org/wiki/Scaling#KeepingtheAspectRatio

**Comment by lukasdotcom** [comment]
I think I'll make another PR that uses the longer side to specify the resolution.

**Comment by lastzero** [comment]
Also be aware that we are still using FFmpeg 5 (included with Ubuntu 23.04). FFmpeg 6 sypports slightly different parameters, from what I know.

**Comment by lukasdotcom** [comment]
I just noticed when I looked at what google photos does in the web video player, that it uses the height to specify the resolution 720p, 1080p, etc. I can still make the PR for it to use the larger side for the resolution or I can just follow what google photos does.

**Comment by lastzero** [comment]
Google might behave differently on a phone though (where they use their native app, which we don't have). So we have to expect the videos to be played directly on a phone, in which case you might otherwise have bad quality. At least that is my understanding.

**Comment by lastzero** [comment]
There's a transcoding test failing in the photoprism package, so that should be fixed as well (make sure to use our official development image for testing or, at a minimum, the same FFmpeg version):

failed

**Comment by lukasdotcom** [comment]
Sorry about not seeing that earlier. I added it really quickly to the PR that makes the resolution use the larger side of the video. #3510

**Pull request merged**